### PR TITLE
Update java-rest-service to use dynamic provisioning

### PR DESCRIPTION
**Pull request by trisberg**
Switch to use dynamic db provisioning

Remove any non-developer yaml files

Remove test pipelines and update README referencing the Namespace Provisioner piplines

### CODE REVIEW

File: java-rest-service/config/testcontainers-test-pipeline.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: Pipeline
 metadata:
   name: java-test-pipeline
```
Comment (gm2552) [inline java-rest-service/config/testcontainers-test-pipeline.yaml:4]: I thought we were removing all test pipeline configuration from the accelerator.

File: java-rest-service/config/workload-basic.yaml [hunk 2]
```diff
@@ -18,7 +18,7 @@ spec:
         autoscaling.knative.dev/minScale: "1"
     - name: testing_pipeline_matching_labels
       value:
-        apps.tanzu.vmware.com/pipeline: test-java
+        apps.tanzu.vmware.com/language: java
```
Comment (gm2552) [inline java-rest-service/config/workload-basic.yaml:21]: Have we confirmed that the namespace provisioner provided Tekton pipelines have removed `apps.tanzu.vmware.com/pipeline: test-java`?  I believe they have, but just checking for verification.  Using only `apps.tanzu.vmware.com/language: java` will not work if the Tekton pipeline did not also removed `apps.tanzu.vmware.com/pipeline: test-java`

File: java-rest-service/config/testcontainers-test-pipeline.yaml (hunk 1 above)
Comment (trisberg) [reply java-rest-service/config/testcontainers-test-pipeline.yaml:4]: it is only in the repo, not part of the generated code

File: java-rest-service/config/workload-basic.yaml (hunk 2 above)
Comment (trisberg) [reply java-rest-service/config/workload-basic.yaml:21]: this is a chicken or egg problem, maybe just leave the pipeline matching out

**Review by gm2552 — APPROVED** [review]

**Pull request merged**